xrpl/models/transactions/oracle_set.py

```diff
@@ -0,0 +1,48 @@
+"""Model for OracleSet transaction type."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import List, Optional
+
+from xrpl.models.nested_model import NestedModel
+from xrpl.models.required import REQUIRED
+from xrpl.models.transactions.transaction import Transaction
+from xrpl.models.transactions.types import TransactionType
+from xrpl.models.utils import require_kwargs_on_init
+
+
+@require_kwargs_on_init
+@dataclass(frozen=True)
+class OracleSet(Transaction):
+    """Represents a OracleSet transaction."""
+
+    account: str = REQUIRED  # type: ignore
+    oracle_document_id: int = REQUIRED  # type: ignore
+
+    """
+    The below three fields must be hex-encoded. You can
+    use `xrpl.utils.str_to_hex` to convert a UTF-8 string to hex.
+    """
+    provider: Optional[str] = None
+    uri: Optional[str] = None
+    asset_class: Optional[str] = None
```

Comment:
Incorrect use of docstrings. This comment will actually be associated with `oracle_document_id`. You'll need to add separate docstrings below each field. Look at other models for reference.

Reply:
fixed in [9c7ce5e](https://github.com/XRPLF/xrpl-py/pull/701/commits/9c7ce5e9e7cabdb5105c193447a2deb27bc6f59e)